### ISSUE TITLE
Can we find out, if a user is already signed in to O365 (angular)?

### ISSUE THREAD

**Issue by bschaeublin** [AzureAD/azure-activedirectory-library-for-js#822]
Im using an [Angular 4+ wrapper](https://github.com/benbaran/adal-angular4), which wrapps the adal-angular@1.0.15.

I'have been searching a long time for a way to find out, if a user can be silenty logged in, or if he would be prompted for credentials after i call `login()` but without any success so far.

This diagram should explain, what im trying to do (green-box):

![image](https://user-images.githubusercontent.com/11610660/47120417-e4b36e00-d26e-11e8-800b-dc8d4bdbc3f7.png)

My way to get to this information was, trying to get a token without login first. My expectation was, that it would return a token, if the user is signed in to O365 in this browser (like, having a valid login.microsoftonline.com cookie).

```
  public async isAbleToSilentLogin(): Promise<boolean> {
    let isLoggedIn = this._adal.userInfo.authenticated;
    if (isLoggedIn) {
      return true;
    }

    try {
      console.log('acquireToken with extraQueryParameter =', this._adal.config.extraQueryParameter);
     // would print  login_hint=test@email.com
      let token = await this._adal.acquireToken(this._adal.config.clientId).toPromise();
      console.log('token', token);
      return !!token;
    } catch (e) {
      console.log(e);
      return false;
    }
  }
```

But all i get is 
> User login is required

Even it should bypass this error, [according to this line](https://github.com/AzureAD/azure-activedirectory-library-for-js/blob/98d7200bc11ebf50405c2dd1a142d99e324c9804/lib/adal.js#L652).

but nvm: I think it is a common scenario, Is there a way to get this information?

**Comment by navyasric** [comment]
@bschaeublin The feature to pass login_hint and then call acquireToken to get silent login is implemented in 1.0.16 release: https://github.com/AzureAD/azure-activedirectory-library-for-js/releases/tag/1.0.16

Please try with the above version.

**Comment by bschaeublin** [comment]
@navyasric thank you for responding. I'm using an [angular2+ wrapper,](https://github.com/benbaran/adal-angular4/issues/91) which still has 1.0.15. And it seems it would not be updated in the next time. 

Do you know any other angular-wrappers using a newer version of adaljs, which works with angular 7 and provides simple interceptor and service classes to use adal?

**Comment by JACrazy** [comment]
version 4 of adal-angular4 has now been released which supports adal-angular 1.0.17
